plugins/services/src/js/structs/PodContainer.js

```diff
@@ -71,21 +71,31 @@ module.exports = class PodContainer extends Item {
   }
 
   hasHealthChecks() {
+    const conditions = this.get("conditions");
+
     // According to RAML specs:
     //
     // https://github.com/mesosphere/marathon/blob/feature/pods/docs/docs/rest-api/public/api/v2/types/container-status.raml#L49
     // 'healthy: should only be present if a health check is defined for this endpoint'
     //
-    return this.get("conditions").some(function(cnd) {
-      return cnd.name === "healthy";
-    });
+    return (
+      conditions &&
+      conditions.some(function(cnd) {
+        return cnd.name === "healthy";
+      })
+    );
   }
 
   isHealthy() {
+    const conditions = this.get("conditions");
+
     // If we have at least 1 health check and it has failed, we are assumed to
     // be unhealthy.
-    return !this.get("conditions").some(function(cnd) {
-      return cnd.name === "healthy" && cnd.value === "false";
-    });
+    return (
+      conditions &&
```

Comment:
If this is undefined, meaning there are no health checks (I guess), this would return false, wouldn't it?

Reply:
not really, empty conditions / undefined means healthy
there's a ancient test that suggests this

Reply:
yes, and `undefined && ...` returns false, making `isHealth()` false. Is my logic flawed here?

Reply:
undefined means `false` and we should return `true` in this case

Reply:
👍 I resolved this by wrapping the conditions check in a `this.hasHealthChecks` call and returning `true` if there are no health checks.